aggregator/rpc_server.go

```diff
@@ -53,24 +63,22 @@ func (r *RpcServer) GetNonce(ctx context.Context, payload *avsproto.NonceRequest
 
 // GetAddress returns smart account address of the given owner in the auth key
 func (r *RpcServer) GetSmartAccountAddress(ctx context.Context, payload *avsproto.AddressRequest) (*avsproto.AddressResp, error) {
-	ownerAddress := common.HexToAddress(payload.Owner)
-	salt := big.NewInt(0)
-	sender, err := aa.GetSenderAddress(r.smartWalletRpc, ownerAddress, salt)
-	nonce, err := aa.GetNonce(r.smartWalletRpc, *sender, salt)
-
+	user, err := r.verifyAuth(ctx)
 	if err != nil {
-		return nil, err
+		return nil, status.Errorf(codes.Unauthenticated, "invalid authentication key")
 	}
+
+	wallets, err := r.engine.GetSmartWallets(user.Address)
+
 	return &avsproto.AddressResp{
-		Nonce:               nonce.String(),
-		SmartAccountAddress: sender.String(),
+		Wallets: wallets,
 	}, nil
 }
 
 func (r *RpcServer) CancelTask(ctx context.Context, taskID *avsproto.UUID) (*wrapperspb.BoolValue, error) {
 	user, err := r.verifyAuth(ctx)
 	if err != nil {
-		return nil, err
+		return nil, status.Errorf(codes.Unauthenticated, "invalid authentication key")
```

Comment:
Let’s capitalize the first word of error messages, "Invalid".

Also, I’d suggest us to create a list of constant for the error message, for code re-use purpose.

Reply:
hmm, is this PR supposed to be on top of #33?